hack/check-everything.sh

```diff
@@ -70,7 +70,7 @@ function fetch_go_tools {
   header_text "Checking for gometalinter.v2"
   if ! is_installed golangci-lint; then
     header_text "Installing golangci-lint"
-    GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.15.0
+    go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.15.0
```

Comment:
I know this isn't changing current behavior, but maybe this tool should be installed with `GO111MODULE=off`? The current version will add this as a dependency to `go.mod` cluttering the file. We don't really want to carry a lint tool as a dependency and someone might accidentally commit it.

Related https://github.com/golang/go/issues/30515

Reply:
I agree that it isn't great to clutter and inadvertantly change `go.mod`. This approach doesn't work though:

```
$ ./hack/check-everything.sh
using tools
Checking for golangci-lint
Installing golangci-lint
go: cannot use path@version syntax in GOPATH mode
```

I updated this to use the `curl`/`goreleaser` installation method that's used elsewhere. Was there some reason we weren't using that here?

Reply:
> The GO111MODULE=off approach doesn't work though

Ah, yes, because of the version. Got it. 

> Was there some reason we weren't using that here?

If I have to guess, it was done before go modules were introduced and it never changed.